modules/earthengine-layers/src/earth-engine-layer.js

```diff
@@ -24,10 +27,21 @@ export default class EarthEngineLayer extends CompositeLayer {
   }
 
   updateState({props, oldProps, changeFlags}) {
+    this._updateToken(props, oldProps, changeFlags);
     this._updateEEObject(props, oldProps, changeFlags);
     this._updateEEVisParams(props, oldProps, changeFlags);
   }
 
+  async _updateToken(props, oldProps, changeFlags) {
+    if (!props.token || props.token === oldProps.token) {
```

Comment:
We may want to make this a bit resilient, i.e. if the user uses multiple EELayers we only want to initialize once.

I would propose storing the token in a global variable (next to `eeApi` above), compare against that. 

Since this logic would be non-standard it would make sense to have a line or two of comments explaining what we are doing.

Reply:
Agreed, I'd be happy to make this global.